mmcls/core/hook/ClassNumCheckHook.py

```diff
@@ -0,0 +1,75 @@
+# Copyright (c) OpenMMLab. All rights reserved
```

Comment:
According to the [naming convention](https://google.github.io/styleguide/pyguide.html#3164-guidelines-derived-from-guidos-recommendations), it's better to name the file as `class_num_check_hook.py`